### PR TITLE
site: add a beginner tutorial for Go

### PR DESCRIPTION
**Pull request by jayconrod**
This augments the beginner tutorials for C++, Java, Android, and iOS.

I kept the structure similar to the C++ tutorial but included some brief information on MODULE.bazel, since that's required for rules_go.

### CODE REVIEW

**Comment by jayconrod** [comment]
Examples PR: bazelbuild/examples#519

**Comment by jayconrod** [comment]
A gentle ping for review. I sent this just before Bazelcon, so I apologize for the awkward timing.

**Review by meteorcloudy — APPROVED** [review]
Thanks!

/cc @fmeum in case you also want to take a look.

File: site/en/start/go.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+$ bazel test //...
+```
+
+## Conclusion and further reading
+
+In this tutorial, we built and tested a small Go project with Bazel, and we learned some core Bazel concepts along the way.
+
+- To get started building other applications with Bazel, see the tutorials for [C++](/start/cpp), [Java](/start/java), [Android](/start/android-app), and [iOS](/start/ios-app).
+- You can also check the list of [recommended rules](/rules) for other languages.
+- For more information on Go, see the [rules_go](https://github.com/bazelbuild/rules_go) module, especially the [Core Go rules](https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md) docuemntation.
```
Comment (fmeum) [inline site/en/start/go.md:342]:
```suggestion
- For more information on Go, see the [rules_go](https://github.com/bazelbuild/rules_go) module, especially the [Core Go rules](https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md) documentation.
```

File: site/en/start/go.md [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+- To learn more about working with modules outside your project, see [external dependencies](/docs/external).
```
Comment (fmeum) [inline site/en/start/go.md:343]: Could make sense to say `Bazel modules` here as Go modules would be fetched with Gazelle.

Could you also mention external deps and reference the rules_go `bzlmod.md` guide for that?

**Review by fmeum — APPROVED** [review]

File: site/en/start/go.md (hunk 2 above)
Comment (jayconrod) [reply site/en/start/go.md]: Thanks, made both changes.

**Comment by jayconrod** [comment]
Thanks for reviewing! https://github.com/bazelbuild/examples/pull/519 has the example code for this, so let me know if you have feedback for that one, too.

File: site/en/start/go.md [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+# Bazel Tutorial: Build a Go Project
+
+{% include "_buttons.html" %}
+
+This tutorial introduces you to the basics of Bazel by showing you how to build a Go (Golang) project. You'll learn how to set up your workspace, build a small program, import a library, and run its test. Along the way, you'll learn key Bazel concepts, such as targets and `BUILD` files.
+
+Estimated completion time: 30 minutes
+
+## What you'll learn
```
Comment (meteorcloudy) [inline site/en/start/go.md:12]: Is there actually missing content for `## What you'll learn`?

Comment (jayconrod) [reply site/en/start/go.md]: Sorry, I meant to take that heading out, now done. I initially followed some of the structure from the Java tutorial, but then it seemed better to just write the things you'll learn in the first paragraph.

Comment (meteorcloudy) [reply site/en/start/go.md]: Np!

**Comment by meteorcloudy** [comment]
I had to make some small tweaks while importing this PR based on https://developers.google.com/style, feel free to send follow up PRs if you think some of the modifications don't make sense.

**Comment by jayconrod** [comment]
Looks good! Thanks again for reviewing and publishing.

**Comment by meteorcloudy** [comment]
Thanks for the nice documentation!

**Pull request closed without merging**